src/main/java/ch/njol/skript/expressions/ExprAge.java

```diff
@@ -0,0 +1,171 @@
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Age of Block/Entity")
+@Description({
+	"Returns the age or maximum age of blocks and age for entities (there in no maximum age for entities).",
+	"For blocks, 'Age' represents the different growth stages that a crop-like block can go through. " +
+	"A value of 0 indicates that the crop was freshly planted, whilst a value equal to 'maximum age' indicates that the crop is ripe and ready to be harvested.",
+	"For entities, 'Age' represents the time left for them to become adults and it's in minus increasing to be 0 which means they're adults, " +
+	"e.g. A baby cow needs 20 minutes to become an adult which equals to 24,000 ticks so their age will be -24000 once spawned."
+})
+@Examples({
+	"# Set targeted crop to fully grown crop",
+	"set age of targeted block to maximum age of targeted block",
+	" ",
+	"# Spawn a baby cow that will only need 1 minute to become an adult",
+	"spawn a baby cow at player",
+	"set age of last spawned entity to -1200 # in ticks = 60 seconds"
+})
+@RequiredPlugins("Minecraft 1.13+")
+@Since("INSERT VERSION")
+public class ExprAge extends SimplePropertyExpression<Object, Integer> {
+
+	static {
+		String fromType = "";
+		
+		if (Skript.classExists("org.bukkit.entity.Ageable"))
+			fromType += "entities";
+		if (Skript.classExists("org.bukkit.block.data.Ageable"))
+			fromType += "/blocks"; // org.bukkit.entity.Ageable exists before org.bukkit.block.data.Ageable
+
+		if (!fromType.isEmpty())
+			register(ExprAge.class, Integer.class, "[:max[imum]] age", fromType);
+	}
+
+	private boolean isMax = false;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		isMax = parseResult.hasTag("max");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Object obj) {
+		if (obj instanceof Block) {
+			BlockData bd = ((Block) obj).getBlockData();
+			if (!(bd instanceof Ageable))
+				return null;
+			Ageable ageable = (Ageable) bd;
+			return isMax ? ageable.getMaximumAge() : ageable.getAge();
+		} else if (obj instanceof org.bukkit.entity.Ageable) {
+			return ((org.bukkit.entity.Ageable) obj).getAge();
+		}
+		return null;
+	}
+	
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (isMax || mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.DELETE)
+			return null;
+		return CollectionUtils.array(Number.class);
+	}
+	
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (mode != ChangeMode.RESET && mode != ChangeMode.DELETE && (delta == null || delta[0] == null))
+			return;
+
+		int newValue;
+		switch (mode) {
```

Comment:
Instead of doing the repeated for loops for each case we can probably just do one with the switch statement within (like we discussed on that one PR)